allow user to control if vault decryption is fatal

bcoca opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->
fixes #13244 #32687
related #37161 #31141
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request


##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
valut
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
2.7
```

AlanCoding commented:
+1

AlanCoding commented on lib/ansible/parsing/vault/__init__.py:
```diff
@@ -657,7 +657,13 @@ def decrypt(self, vaulttext, filename=None):
+            if C.ERROR_ON_VAULT_FAIL:
+                raise
+            else:
+                display.warning('Vault decryption failed: %s' % to_native(e))
```
Now I'm wondering, what will be the value of `plaintext` in this scenario, and what the implications of that will be.

AlanCoding commented:
I've confirmed my hunch.

You can see that both cases still fail. It doesn't really accomplish a pass-through if the setting is set to `False`, as one would expect. There is a different message emitted in both cases, so your conditional really is intercepting the missing secrets case, but there appears to be more cases that also need to be handled.

I would still like to have this, I would be happy to incorporate use of this constant into my other PR.

##### false case
```
$ ANSIBLE_ERROR_ON_VAULT_FAIL=false ansible-playbook -i scripts/vault/awx_redumper.py debugging/hostvars_print.yml

PLAY [all] ****************************************************************************************************************************************************************************

TASK [debug] **************************************************************************************************************************************************************************
 [WARNING]: Vault decryption failed: Attempting to decrypt but no vault secrets found

fatal: [foobar]: FAILED! => {"msg": "An unhandled exception occurred while templating '{{ hostvars[inventory_hostname] }}'. Error was a <type 'exceptions.UnboundLocalError'>, original message: local variable 'plaintext' referenced before assignment"}
	to retry, use: --limit @/Users/alancoding/Documents/repos/ansible-inventory-file-examples/debugging/hostvars_print.retry

PLAY RECAP ****************************************************************************************************************************************************************************
foobar                     : ok=0    changed=0    unreachable=0    failed=1   

```

##### true case
```
$ ANSIBLE_ERROR_ON_VAULT_FAIL=true ansible-playbook -i scripts/vault/awx_redumper.py debugging/hostvars_print.yml

PLAY [all] ****************************************************************************************************************************************************************************

TASK [debug] **************************************************************************************************************************************************************************
fatal: [foobar]: FAILED! => {"msg": "An unhandled exception occurred while templating '{{ hostvars[inventory_hostname] }}'. Error was a <class 'ansible.parsing.vault.AnsibleVaultError'>, original message: Attempting to decrypt but no vault secrets found"}
	to retry, use: --limit @/Users/alancoding/Documents/repos/ansible-inventory-file-examples/debugging/hostvars_print.retry

PLAY RECAP ****************************************************************************************************************************************************************************
foobar                     : ok=0    changed=0    unreachable=0    failed=1  
```

bcoca commented:
might need to change then, ignore_missing_secrets vs ignore_failed_decrypt, probably need the 2nd toggle a few steps up.

bcoca commented:
@AlanCoding i updated to both cover all types of errors and make the distinction between missing/mismatched secrets vs other general vault errors.

alikins commented on lib/ansible/config/base.yml:
```diff
@@ -1244,6 +1244,22 @@ ERROR_ON_MISSING_HANDLER:
+ERROR_ON_VAULT_FAIL:
+  name: Error on vault decryption failure
+  default: True
+  description: "Toggle to speicfy if 'non-secret' related vault decryption failures are errors or warnings."
```
s/speicfy/specify

alikins commented on lib/ansible/config/base.yml:
```diff
@@ -1244,6 +1244,22 @@ ERROR_ON_MISSING_HANDLER:
+ERROR_ON_VAULT_SECRET:
+  name: Error on missing/mismatched vault secrets
+  default: True
+  description: "Toggle to speicfy if vault decryption caused by missing or mismatch secrets to be errors or warnings."
```
s/speicfy/specify

AlanCoding commented:
There's a much better solution to the particular use case I was interested in via 

https://github.com/ansible/ansible/issues/37536
https://github.com/ansible/ansible/issues/37531

The particular behavior of "don't decrypt vault values" can reasonably be accomplished with a simple flag on `ansible-inventory`

kenyon committed:
complex_data_manipulation.rst: fix typo (#79758)

felixfontein committed:
Argument spec alias handling: actually report deprecated aliases in suboptions, and fix warning message in suboptions when two aliases of the same option are used (#79740)

* Normalize deprecation records.

* Fix alias deprecations in suboptions.

* Report in which option an alias warning happened for suboptions.

* Add deprecation tests for suboptions.

* Also test deprecation in list of dicts.

* Adjust unit tests for toplevel alias deprecation field name change.

kristianheljas committed:
Fix broken link for inventory script examples (#79762)

These scripts were removed from community.general in https://github.com/ansible-collections/community.general/pull/2696 and moved to ansible-community/contrib-scripts per https://github.com/ansible-community/community-topics/issues/16#issuecomment-853317111

Andersson007 committed:
communication.rst: improve (#79751)

mattclay committed:
Fix broken docs links. (#79769)

mdavis-xyz committed:
document where to save cloud config files (#79412)

mattclay committed:
Flush output in sanity requirements update script. (#79774)

This keeps the subprocess output under the correct heading.

mkrizek committed:
Fix traceback in template action with ANSIBLE_DEBUG=1 (#79764)

Fixes #79763

mattclay committed:
ansible-test - Remove obsolete DirectoryTarget.

This code has been unused since at least the 2.9 release.

samccann committed:
fix filename for sidecar docs (#79779)

mattclay committed:
ansible-test - Deprecate the govcsim container.

Also remove the integration tests for the `vcenter` test plugin, since the tests relied on using the simulator.

mattclay committed:
ansible-test - Fix various type hinting issues. (#79798)

* ansible-test - Add missing type hints.

* ansible-test - Remove redundant type hints.

* ansible-test - Fix return type annotations.

* ansible-test - Add assert, casts to assist mypy.

* ansible-test - Fix incorrect type hints.

* ansible-test - Remove no-op code.

* ansible-test - Fix incorrect types.

* ansible-test - Fix method signature mismatch.

mkrizek committed:
Fix conditionally notified include handlers (#79804)

Fixes #79776

ci_complete

mattclay committed:
validate-modules - Remove `__future__` limits (#79800)

* validate-modules - Remove `__future__` limits

Limits on specific `__future__` imports are handled by other sanity tests.

* Add integration test for module/plugin imports.

mattclay committed:
Use `raise ... from` instead of `six.raise_from`

evgeni committed:
correct examples to use removed_from_collection not collection_name (#79803)

Andersson007 committed:
[Docs] maintainers_guidelines: add WG and real-time chat request info (#79750)

felixfontein committed:
Bump acme test container. (#79783)

mattclay committed:
Drop requirement for matching sanity constraints

Each sanity test uses its own virtual environment, so there is no risk of conflicting requirements between tests.

It may still be a good idea to make an effort to keep frozen requirements in sync between tests.

Making it a hard requirement unnecessarily complicates updating individual tests and increases the risk of incorrect manual changes.

prasadpatil49 committed:
doc fix for platform content #79794 (#79801)

991jo committed:
Fix: documentation for per-task timeout (#79715)

mattclay committed:
ansible-test - Update CloudStack plugin container (#79813)

bcoca committed:
strategy fix invalid variables being registered (#79706)

also added tests for normal and bad cases

mattclay committed:
Update mypy and its dependencies. (#79812)

* Update mypy and its dependencies.

* Fix requirements.

* Remove unnecessary matching constraints.

mattclay committed:
ansible-test - Update sanity test requirements (#79819)

Also update core sanity test requirements, except mypy (already updated) and docs-build (update separately).

mattclay committed:
ansible-test - Update the NIOS test container (#79820)

mattclay committed:
ansible-test - Deprecate the foreman test plugin (#79821)

s-hertel committed:
fix installing collections from git repos that contain MANIFEST.json (#79808)

* add null 'manifest' key to metadata for git repo collections containing MANIFEST.json

changelog

* set to Sentinel instead of None

* Test installing a collection in a git repo that contains a MANIFEST.json

* fix test

* Update changelogs/fragments/ansible-galaxy-install-git-src-manifest.yml

mattclay committed:
Expand docs for the import sanity test. (#79768)

* Expand docs for the import sanity test.

* Remove note about Python 2.7 compat.

It should not be needed since there is a sanity test to enforce use of `__metaclass__ = type`.

* Improve introductory paragraph.

* Fix link typo.

Hofer-Julian committed:
docs: Extend password entry of ansible.builtin.user (#79694)

* docs: Extend password entry of ansible.builtin.user

Clarify that `password` sets the password hash.
Not the actual password.
Fixes part of  #79684

Commit:
Update dev_guide.rst (#79625)

ssbarnea committed:
Improve documentation on requirements.yml (#76140)


Makes it clear that user can use range identifiers with collection
versions inside requirements.yml files.

Andersson007 committed:
[Docs] add easyfix/good first issue/docs links (#79830)

Andersson007 committed:
[Docs] add doc links to documentation_contributions.rst (#79840)

mariolenz committed:
Remove dev_guide stubs (#79795)

* Remove dev_guide stubs

* Remove Cisco ACI Dev Guide

mattclay committed:
Ansible.Basic.cs - Fix compile error on PS 7.3.x (#79848)

mattclay committed:
ansible-test - Update base and default containers (#79846)

mattclay committed:
Remove unused and unreachable unit test code (#79854)

jborean93 committed:
Add tests to cover win_reboot incidental paths (#79856)

* Add tests to cover win_reboot incidental paths

* Fix sanity issues

jborean93 committed:
AddType - Support compiling with /unsafe C# code (#79853)

* AddType - Support compiling with /unsafe C# code

* Update Ansible version in docstring

rooftopcellist committed:
Add Ansible community 7.2.0 porting guide (#79867)

Andersson007 committed:
maintainers_guidelines.rst: add a link to collection release guidelines (#79859)

mattclay committed:
Remove outdated note on relative imports in docs (#79873)

mattclay committed:
ansible-test - Update pylint to 2.16.0 (#79878)

mattclay committed:
Remove unused unit test code (#79879)

* Remove unused test fixtures

* Removed unused _old_dump_load_cycle method

* Remove Python 2.x compat

* Remove unused code

* Remove unused context manager

* Fix cowsay test

- The test no longer depends on another test to initialize config.
- Also remove unreachable code.

* Remove Python 2.x compat

mattclay committed:
ansible-test - Specify config path in plugin error (#79881)

mattclay committed:
ansible-test - Update default containers (#79887)

felixfontein committed:
docsite: bump antsibull-docs, reference collection env var index (#79666)

JaroslavKlech committed:
Mentions 'meta: flush_handlers' task (#79542)

Dule-martins committed:
Description for changing User ID to match user value (#79470)

Commit:
Update dnf.py (#79679)

kgawda committed:
Remove irrelevant line (#79865)

Remove irrelevant comment line form example code

sivel committed:
Use arg validation in debug action (#79866)

* Use arg validation in debug action. Fixes #79862

* Add clog frag

* Move the msg default into the argspec

mattclay committed:
user - Remove invalid return docs (#79891)

mattclay committed:
Clean up unused imports in unit tests (#79893)

mattclay committed:
Clean up unused imports in `hacking` directory (#79894)

mattclay committed:
Clean up unused imports in integration tests (#79895)

mattclay committed:
Clean up unused imports in sanity tests (#79896)

mattclay committed:
Clean up unused imports in ansible-test unit tests (#79897)

mattclay committed:
Clean up unused imports in modules (#79898)

mattclay committed:
Remove unused internal imports from module_utils (#79906)

mattclay committed:
Remove unused stdlib imports from module_utils (#79905)

mattclay committed:
Clean up unused imports in core (#79900)

* Clean up unused imports in core

* Add changelog fragment

mattclay committed:
Clean up unused imports in plugins (#79899)

mattclay committed:
Remove unwanted comma from pause action (#79908)

mattclay committed:
Ignore necessary unused imports in module_utils (#79904)

mattclay committed:
ansible-test - Enable trailing-comma-tuple pylint rule (#79909)

mattclay committed:
ansible-test - Enable pylint unused-import for core (#79910)

mattclay committed:
Update collections.abc imports (#79911)

* Update `collections.abc` imports

- Use `six.moves` for modules and module_utils
- Use `collections.abc` for controller code

This avoids using `ansible.module_utils.common._collections_compat`,
which was added before the vendored `six` was updated to provide these
imports.

* Update _collections_compat to use six.moves

Also update the custom pylint rule to reflect this change.

s-hertel committed:
Remove another unwanted comma from pause action (#79926)

mattclay committed:
ansible-test - Remove old containers and remotes (#79929)

mateusrangel committed:
Improving the RETURN and its docs on the apt_repository module (#79658)

* Improving the documentation on how we generate the default value of the filename parameter

* fix pep8

* removing unnecessary documentation and improving the module's return

* making the RETURN docs

* pep8

* version_added and changelog

* module._diff

* module._diff fix

* add rudimentary tests for new outputs

---------

Co-authored-by: Matt Davis <mrd@redhat.com>

sivel committed:
Update example to always init ANOTHER_LIBRARY_IMPORT_ERROR (#79927)

acozine committed:
make this a warning, not a comment (#79928)

Co-authored-by: Alicia Cozine <acozine@users.noreply.github.com>

nitzmahone committed:
quiet default ansible-doc integration test output (#79921)

* quiet default ansible-doc integration test output 

* typical non-verbose output was exceeding 27k lines per fun
* disables `set -x` unless `-v` is passed to the script (eg when ansible-test is called with `-v` or `--retry-on-error` adds it on the second try
* added simple progress echoes
* suppress some grep output

* only enable `set -x` for >= `-vvv`

* fix shellcheck default complaint

mattclay committed:
ansible-test - Fix file permissions for delegation (#79932)

* ansible-test - Fix file permissions for delegation

* Set more restrictive permissions for SSH key

* Check all execute bits, not just owner

* Add a breaking_changes changelog entry

s-hertel committed:
include_vars - document hash_behaviour + dir (#79790)

* document that hash_behaviour does not apply to individual files of variables when using the dir option

* add tests

mattclay committed:
Fix pointless statements in unit tests (#79940)

mattclay committed:
Clean up pointless statements (#79941)

* Add ignores for intentional pointless statements

* Remove redundant docstring

* Fix function docstring placement

mattclay committed:
Fix "pointless" strings detected by pylint (#79943)

Convert these strings to comments, since they are not valid docstrings.

mattclay committed:
Enable pylint rules to detect pointless statements (#79944)

Also instruct `git blame` to ignore the commit used to convert pointless strings to comments.

felixfontein committed:
ansible-test - Fix collection delegation (#79947)

samccann committed:
Rst fixes (#79939)

* fix rstcheck errors for 6.1.1
* spell shell correctly

mattclay committed:
Fix detection of available hashlib algorithms (#79946)

* Fix detection of available hashlib algorithms

Detection of hashlib algorithms now works on Python 3.x.
The new implementation works on Python 2.7 and later.

Test coverage is provided by both integration and unit tests.

* Add additional details about hashlib in docs

s-hertel committed:
improve password_hash warning for unsupported algorithms (#79872)

* password_hash - give a warning for unsupported algorithms (that raise a TypeError)

* add suggested changes, a test and changelog

mattclay committed:
pycompat24 - Remove support for Python <= 2.5 (#79953)

mattclay committed:
basic.py - Perform Python version check earlier (#79954)

nitzmahone committed:
don't ignore templated _raw_params that k=v parser failed to parse (#79913)

fixes #79862

mkrizek committed:
Change ansible_job_id format (#79951)

* Change ansible_job_id format

...to be something that does not resemble float or other type which
value could be changed by literal_eval that jinja2_native uses.

Specifically the format of '%d.%d' is converted from str to float and
then back to float which may result in truncating the number resulting
in the job not being found because the job id does not exist.

sivel committed:
Replace deprecated stacked `@classmethod` and `@property` (#79952)

mkrizek committed:
Remove unneeded fallback code for nested defined/undefined tests (#79960)

* Remove unneeded fallback code for nested defined/undefined tests

The code used to handle defined/undefined tests on objects containing
nested undefined variables. Due to changes in lazy evalution of Jinja2
expressions it is no longer needed, see #56116.

rndmh3ro committed:
Change `yes` to `true` in uri-module (#79934)

tremble committed:
Add an integration test to ensure that defaults don't count towards mutually exclusive (#79949)

samccann committed:
fix core devel doc output (#79902)

mattclay committed:
Update mypy to version 1.0.0 (#79963)

mattclay committed:
ansible-test - Clean up indentation and spaces (#79980)

bcoca committed:
ansible-inventory, add --limit option (#79596)

* ansible-inventory, add --limit option

* also graph

* optimize the when not providing limit

* added tests and clog

* avoid empty groups, fix tests

* i swear there as an ignore_errors there already!!?!?!?

* Fix stdout test fallback

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>

* simpler approach using subset against inventory directly

* tyupose

* renamed funciton to what it actuall does
got yaml in line with others about removing empty groups

* have graph ignore limts, also note same for --host

* fixed typo long line

* better test

* cause hosvars are not a thing

---------

Co-authored-by: Matt Davis <6775756+nitzmahone@users.noreply.github.com>
Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>

mattclay committed:
ansible-test - Remove auto format pylint rule (#79985)

mattclay committed:
ansible-test - Enable unused-import pylint rule (#79912)

This was previously enabled for core. This change enables it for collections.

mattclay committed:
ansible-test - Improve code formatting (#79983)

* ansible-test - Add blank lines after docstrings

* ansible-test - Preserve formatting of arg pairs

* ansible-test - Remove unused string

* ansible-test - Remove pointless dict() usage

* ansible-test - Clean up initial func arg indenting

* ansible-test - Clean up constructor arg indenting

* ansible-test - Clean up func arg wrapping

* ansible-test - Clean up comma and paren placement

s-hertel committed:
user - fix comparing existing group names to group IDs (#79981)

acozine committed:
adds  roadmap for Ansible 8 (#79598)

acozine committed:
fixes typo in Ansible 7 roadmap (#79996)

Co-authored-by: Alicia Cozine <acozine@users.noreply.github.com>

mattclay committed:
Disable F37 podman tests for ansible-test (#79998)

The recent update from podman 4.3.1 to 4.4.1 has broken container management.
Removing the tests from the CI matrix until the issue is resolved.

mattclay committed:
ansible-test - Support Podman 4.4.0+

mattclay committed:
ansible-test - Improve pylint backwards compat (#79997)

mattclay committed:
Revert "Disable F37 podman tests for ansible-test (#79998)"

This reverts commit 875a4ec2428f289cbeb855ccb4e099e67b863813.

mattclay committed:
ansible-test - Disable the pep8 E203 rule (#79966)

The `E203` rule in `pycodestyle` is not fully PEP 8 compliant.
Disabling it provides compatibility with the output of the `black` code formatter.

mkrizek committed:
Make using blocks as handlers a parser error (#79993)

Fixes #79968

jborean93 committed:
Add condition that causes a when to skip a task to output msg (#78918)

* Add condition that causes a when to skip a task

* Fix up tests

* Use false_condition instead of failed_condition

* Remove formatting accidentially added

* Fix sanity

sivel committed:
Data tagging delayed to 2.16 (#80011)

samccann committed:
update banner' (#79971)

samccann committed:
Update docs requirements to latest package versions (#79930)

Co-authored-by: Matt Clay <matt@mystile.com>

mattclay committed:
Add missing newlines to YAML files (#80031)

mattclay committed:
Remove trailing spaces from lines in YAML (#80032)

mattclay committed:
Avoid trailing spaces in unit test (#80033)

msagi committed:
ansible-playbook -K breaks when passwords have quotes (#79837)

Now only 'unquote' when ini config file is the source

anzoman committed:
Add Steampunk Spotter to Other tools list (#80006)

devon-mar committed:
Update lookup to use positional only arg for name to avoid conflicts (#79839)

mattclay committed:
ansible-test - Support pylint assertion rewriting (#80020)

Add support for `pylint` assertion rewriting when running unit tests on Python 3.5 and later.

mattclay committed:
ansible-test - Allow target prefixes to use `_` (#80021)

Integration test target prefixes defined in a `tests/integration/target-prefixes.{group}` file can now contain an underscore (`_`) character.

mattclay committed:
ansible-test - Show Python version before install (#80022)

Always indicate the Python version being used before installing requirements.

mattclay committed:
ansible-test - Update mypy and pylint requirements (#80049)

webknjaz committed:
Fix #80020 typos confusing `pylint` for `pytest` (#80056)

mattclay committed:
ansible-test - Update base and default container (#80058)

mattclay committed:
ansible-test - Adjust vcenter test plugin behavior (#80063)

The previous change to the default mode of operation for the plugin is now limited to collections.

This enables easier testing of the plugin in core, so the previously removed tests have been restored.

oraNod committed:
add wheel to docsite pre-reqs #79513 (#80061)

devon-mar committed:
Fix markdown in changelog fragment (#80065)

mattclay committed:
Update ansible-test openshift plugin tests (#80064)

Update ansible-test openshift plugin tests to use config to get server endpoint.

Also update the comment explaining why the tests are disabled.

Gijikiki committed:
Update playbooks_tests.rst (#79788)

mkrizek committed:
Use FAs with inheritance only when applicable (#80026)

... and set default value of an attribute on an object
only in NonInheritableFA.

Fixes #79777

ci_complete

nkakouros committed:
Clarify when plugin settings are being populated (#79710)

Player256 committed:
Update collection_integration_updating.rst (#80037)

aasthavar committed:
consistent booleans for page text and code blocks (#80036)

* consistent booleans for page text and code blocks

* Update check_mode from n to false

DesmondSanctity committed:
fix: jinja complex type transforms doesn't work as documented (#80067)

jinja complex type transforms: dict(somelist| slice(2)) doesn't work as documented. Changed the example to use zip_longest

DesmondSanctity committed:
package_facts requires python-rpm on SUSE systems in ansible 2.12.1 (#80041)

mkrizek committed:
Introduce Delegatable and Notifiable mixin classes (#80077)

* Introduce Delegatable and Notifiable mixin classes

* fix sanity

* fix sanity

raphaelm committed:
Docs: Clarify note on tags (#80079)

The current wording could be read as "tasks that are tagged *both* configuration and packages" while in fact it is "tasks that are tagged *either* configuration *or* packages"

webknjaz committed:
📦 Integrate manpage builds into PEP 517 build backend

This patch creates a thin wrapper around the `setuptools`' PEP 517
build backend in-tree. It features an ability to request generating
the manpage files in the process of building a source distribution.
This toggle is implemented using the `config_settings` mechanism of
PEP 517.
One must explicitly pass it a CLI option to the build front-end to
trigger said behavior. The packagers are expected to use the
following call:

    python -m build --config-setting=--build-manpages

This option has no effect on building wheels.

🧪 The change includes integration tests

This test runs building and re-building sdists and wheels with and
without the `--build-manpages` config setting under the
oldest-supported and new `setuptools` pinned.

It is intended to preserve the interoperability of the packaging setup
across Python runtimes.

An extra smoke test also verifies that non PEP 517 interfaces remain functional.

PR #79606


Co-authored-by: Matt Clay <matt@mystile.com>

mattclay committed:
ansible-test - Fix vendoring support (#80074)

- Support loading of vendored Python packages.
- Exclude vendored Python packages from payloads.

mattclay committed:
Remove unused network test support files (#80080)

* Remove unused network test support files

* Remove obsolete ignores

mattclay committed:
Remove unused integration test support files (#80081)

* Remove unused integration test support files
* Remove obsolete ignores

ci_complete
ci_coverage

mattclay committed:
Remove straight.plugin dependency (#80084)

yselkowitz committed:
Define all processor facts on s390x (#19755) (#79806)

Signed-off-by: Yaakov Selkowitz <yselkowi@redhat.com>

mattclay committed:
ansible-test - Update base and default container (#80086)

mattclay committed:
Relocate validate-modules unit tests (#80087)

This allows the validate-modules unit tests to run as part of the regular unit test suite.

agowa338 committed:
Clarify combine filter examples (#80078)

tremble committed:
sanity tests - ignore pre-release versions for deprecation comparisons (#79185)

sivel committed:
Do not use hardcoded httpbin.org in uri tests (#80101)

anweshadas committed:
Add Ansible community 7.3.0 porting guide (#80106)

webknjaz committed:
🎨 Convert RST to manpage in-memory @ PEP 517 (#80098)

* Add _convert_rst_in_template_to_manpage arg types

* 📦 Make manpage build dependencies conditional

Previously, said dependencies were declared as unconditionally
required even when manpages not needed to be built. This patch
Makes it so they are only required when needed.

* Correct _generate_rst_in_templates returned type

It was marked as Path before this patch but in fact, it's iterable of
paths.

* 🎨 Convert RST to manpage in-memory @ PEP 517

Previously, the automation was writing a temporary templated RST on
disk and calling a helper CLI script on that. But with this change, it
happens with less unnecessary I/O.

Co-Authored-By: Matt Davis <6775756+nitzmahone@users.noreply.github.com>

* 📦Expose sdist manpage build deps unconditionally

Due to a bug in pypa/build, the `get_requires_for_build_sdist()` hook
is always invoked with `config_settings=None`. This means that we
cannot conditionally extend build requirements in said hook.

As a workaround, this patch makes hook pretend that `--built-manpages`
is always passed.

Ref: https://github.com/pypa/build/issues/559.

---------

Co-authored-by: Matt Davis <6775756+nitzmahone@users.noreply.github.com>

mattclay committed:
Relocate and refactor ansible-test diff unit tests (#80113)

This provides the same test coverage as the previous tests, without the dependency on git.
It also includes many more specific test cases with assertions, instead of simply relying on the code to not raise an exception.

s-hertel committed:
add a worker queue to get updates from the main results thread (#79886)

* Create a queue per WorkerProcess to receive intra-task updates
* Update `pause` action to use the worker queue
* Deprecate ConnectionBase()._new_stdin
* Add new `Display`  convenience method `prompt_until` to manage both controller- and worker-sourced prompting without cross-fork stdin sharing, in-worker mechanism to handle request-response over new worker queue.

mattclay committed:
Relocate the AZP entry point scripts (#80114)

Scripts previously under `test/utils/shippable/` are now under `.azure-pipelines/commands/` instead.

samdoran committed:
unarchive - properly handle relative path for `dest` (#75267)

* unarchive - properly handle relative path for dest

* Add integration test

* Return output of underlying commands with increased verbosity

* Revert "Return output of underlying commands with increased verbosity"

This reverts commit a2790c8275cdc5697b65670a0beffdc74b741bf6.

* Warn when a relative destination path was provided

samdoran committed:
unarchive - log errors from underlying commands (#75329)

* unarchive - log errors from underlying commands

When a command run by the module fails, log the errors if debugging is enabled.

* Use debug() method

webknjaz committed:
📦 Drop unnecessary wheel PEP 517 build dependency (#80096)

Code snippets advertising the `wheel` dependency in the
`pyproject.toml`'s `[build-system].requires` setting were a historical
mistake. This has been corrected in
https://github.com/pypa/setuptools/commit/f7d30a95 but many pyprojects
still have `wheel` in their configs.

It is not needed for building sdists and the corresponding setuptools'
PEP 517 hook that provides requirements for building wheels already
esposes this dependency automatically.

oraNod committed:
improve doc dependency install steps (#80120)

0kyn committed:
Fix typo docs in playbooks_tags.rst (#80117)

kristianheljas committed:
Encourage true/false for module documentation (#80060)


Co-authored-by: Felix Fontein <felix@fontein.de>

akira6592 committed:
fix link for development guidelines (#80055)

agowa338 committed:
improve password lookup syntax in examples (#80092)

deining committed:
Fixing typos (#80045)

samccann committed:
clarify  playbook tests (#80073)

Keeper-of-the-Keys committed:
Update playbooks_conditionals.rst (#80062)

"changed" was missing from the docs

webknjaz committed:
🐛 Make integration tests compatible w/ modern Git (#80122)

* 🐛 Make integration tests compatible w/ modern Git

This patch makes use of the `init.defaultBranch` setting to unify
the test across new and old Git versions since one defaults to
`master` and the other uses `main` for the default branch.

Where possible, it uses the `HEAD` committish to avoid having to
normalize the branch name.

The change fixes the following integration tests:

  * `ansible-galaxy`

  * `ansible-galaxy-collection-scm` (recursive collection)

  * `git`

* 🐛Replace `git-symbolic-ref` with a repo template

This custom Git repository template emulates the `init.defaultBranch` setting
on Git versions below 2.28. Ref: https://superuser.com/a/1559582.
Other workarounds mentioned there, like invoking
`git symbolic-ref HEAD refs/heads/main` after each `git init` turned
out to have mysterious side effects that break the tests in surprising ways.

* 🎨 Make Git integration test non-destructive

This patch makes use of the `$HOME` environment variable to trick Git
into using a user-global config generated in the temporary directory.

webknjaz committed:
Add macOS 13 to CI and ansible-test (#79508)

* ✨ Add macOS 13.2 to `ansible-test`

* 🧪 Replace macOS 12.0 with 13.2 in the CI matrix

* Skip `lookup_url` under macOS 13.2

This is due to https://wefearchange.org/2018/11/forkmacos.rst.html
that manifests itself as follows:

    TASK [lookup_url : Test that retrieving a url works] ***************************
    objc[15394]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
    objc[15394]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in t
    he fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
    ERROR! A worker was found in a dead state

* 📝 Extend ansible-test change note w/ macOS 13.2

yajo committed:
docs: automatic `always` tag in gather_facts (#80052)

oraNod committed:
docsite requirements path (#80134)

bcoca committed:
ansible-inventory prevent duplicating host entries (#80059)

this happened after implementing limits we introduced a bug that a host would be duplicated if it existed in a group's children

bcoca committed:
apt_repository, dont assume only missing apt-key will populate stderr (#79827)

  fixes #79825

webknjaz committed:
🧪 Switch macOS 13.2 to 12.0 in CI (#80145)

* 🧪 Switch macOS 13.2 to 12.0 in CI

The former revealed unexpected flakiness while the latter is the
previous value that was used to be stable. This is a temporary revert.

* Skip lookup_url integration test under macOS 12.0

mattclay committed:
Fix MANIFEST.in includes (#80147)

mattclay committed:
ansible-test - Fix pylint error with old home dir (#80151)

mattclay committed:
ansible-test - Fix error with missing pylint home (#80155)

zpeschke committed:
Fix spelling and grammar in subelements filter doc (#80137)

mkrizek committed:
Avoid redundant unsafe wrapping in ansible_eval_concat (#80143)

The wrap_var is done in do_template for all concat functions after their
return.

bcoca committed:
paramiko connection now uses the correct source of data (#79704)

* paramiko connection now uses the correct source of data

* keep orig name

* added missing port to docs

* Update changelogs/fragments/paramiko_config.yml

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>

* test ansible_host templating

* added missing private_key_file

* removed all play_context fallbacks

* Update lib/ansible/plugins/connection/paramiko_ssh.py

Co-authored-by: Jordan Borean <jborean93@gmail.com>

* Update lib/ansible/plugins/connection/paramiko_ssh.py

Co-authored-by: Jordan Borean <jborean93@gmail.com>

---------

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>
Co-authored-by: Jordan Borean <jborean93@gmail.com>

s-hertel committed:
update paramiko_ssh unit tests load the connection with plugin loader (#80157)

* Load the paramiko_ssh connection plugin properly using plugin loader so it can use config manager when _connect() is called

s-hertel committed:
move test so it runs when the plugin is modified (#80158)

webknjaz committed:
Revert "🧪 Switch macOS 13.2 to 12.0 in CI (#80145)" (#80156)

This reverts commit 326e70c02a2563e13a68565c31ce42477ef4874f.

mkrizek committed:
Raise an error on invalid FA.isa value (#80040)

Avoids bad definitions of playbook classes

mattclay committed:
ansible-test - Remove pointless comparison (#80168)

mattclay committed:
ansible-test - Update pylint requirements (#80172)

s-hertel committed:
copy - fix check mode with remote_src=True (#78624)

* Don't create dest directory in check mode

uncomment existing test

Fix checking for file attribute changes in check mode and add a test

mattclay committed:
Update mypy to 1.1.1 (#80175)

mattclay committed:
ansible-test - Update base and default container (#80178)

harmdhast committed:
Fixed typo (#80184)

`uvault` should be `unvault`

s-hertel committed:
ansible-galaxy role info - fix unhandled AttributeError

* catch GalaxyError instead of its base class to ensure it will have a http_error attribute

kristianheljas committed:
Adjust release schedule docs to four weeks (#80166)

hugovk committed:
Replace deprecated BadZipfile with BadZipFile (#80198)

* Replace deprecated BadZipfile with BadZipFile

* Retain support for Python 2.7

mattclay committed:
ansible-test - Update git diff handling (#80202)

This change allows ansible-test to work with newer versions of git on AZP.

eltociear committed:
fix typo in plugin_docs.py (#80195)

lable -> label

sumanth-lingappa committed:
corrected class documentation (#80241)

acozine committed:
Document debugging conditionals (#80239)

##### SUMMARY
Add a section to the docs describing how to debug conditional statements - how to get Ansible to show you whether your `when:` clause evaluates to `true` or `false`.

I ran into trouble with this and couldn't find anything in the docs. Thought I'd add it.


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

EstherChristopher committed:
Update connection.rst (#80194)

ktdreyer committed:
uri: improve force_basic_auth documentation (#80211)

Add more details about what "true" and "false" mean for the
force_basic_auth setting. Give example scenarios when clients may want
to use this setting.

b-enoit-be committed:
Update special_variables.rst (#80210)

The usage of a glossary on this page will allow linking to a specific special variable, like on the glossary page, e.g.: https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Idempotency

webknjaz committed:
Make isolated source changes @ PEP 517 backend (#80253)

webknjaz committed:
📦 Switch sdist build-system to pure setuptools (#80255)

This patch modifies the in-tree build backend to build sdists that swap
out pointers to it in the `pyproject.toml`'s `[build-system]` section.

The effect of this is that the first build from source (for example,
from a Git checkout) uses our PEP 517 in-tree build backend. But the
produced tarball has `build-backend` set to `setuptools.build_meta`
which is the native build backend of `setuptools`. So any following
builds from that sdist will skip using the in-tree build backend,
calling the setuptools' one.
The good news is that if the first build generated the manpages, they
will be included and won't go anywhere even though, a different build
system is in place.

Combined with #80253, this will make sure not to modify the current
source checkout on that first build.

Co-authored-by: Matt Clay <matt@mystile.com>

webknjaz committed:
Make PEP 517 mutation tests use pinned old setuptools (#80262)

* Pin setuptools to lowest supported @ PEP 517 test

This allows catching the behavior of builds under old setuptools.

* Stop invoking `setup.py install` in tests

This is not the part we care about since it involves dealing with the
external runtime dependencies rather than building our source
distribution.

felixfontein committed:
Bump antsibull-docs version to 1.10.0 to support semantic markup. (#80201)

felixfontein committed:
Implement semantic markup support for Ansible documentation in ansible-doc. (#80242)

samccann committed:
Move Collection requirements to ansible/ansible (#80234)

sivel committed:
ansible-galaxy collection install retry improvements (#80180)

* clog frag

* Fix retries so that each explicit call to _call_galaxy is retried for the correct number of attempts. Fixes #80174

* Extend retry logic to common URL related connection errors. Fixes #80170

* Extend retries to downloading artifacts

* Extend param docs for change

* Rework the exception handling

* Don't be overly broad, reduce to TimeoutError, and BadStatusLine for now

* _download_file needs to raise AnsibleError.orig_exc

* Remove unused import

* Add IncompleteRead

* Add socket.timeout for py39

* Add 502 to retry codes

* Move http error code checking first

* Use itertools.tee to replay the backoff_iterator instead of using a callable

* Actually set a CLI default of 60s for timeout, to prevent implicit galaxy from using 10s as default from Request.open

* Import typing

* fix type hints

* Use http.HTTPStatus instead of int HTTP error codes where feasible

* Split exception handling

Co-authored-by: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>

* Add missing import

---------

Co-authored-by: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>

mattclay committed:
Remove test related targets from Makefile (#80115)

Invoke `ansible-test` directly instead of wrapping invocations with `make`.

mattclay committed:
ansible-test - Update sanity test requirements (#80279)

mattclay committed:
ansible-test - Update CloudStack test container (#80278)

felixfontein committed:
validate-modules: support plugin see-also (#80244)

felixfontein committed:
validate-modules: check_mode attribute and compare with supports_check_mode parameter (#80090)

Co-authored-by: Brian Scholer <1260690+briantist@users.noreply.github.com>

felixfontein committed:
ansible-doc: fix broken seealso links in text output (#80280)

* Fix broken URLs.

* Also remove auto-generated description for modules outside ansible.builtin.

sivel committed:
Do not double calculate loops and `delegate_to` (#80171)

sivel committed:
Don't double loop, ensuring that machine formats don't dupe collections onto all paths (#80277)

* Don't double loop, ensuring that machine formats don't dupe collections onto all paths

* Add tests

mattclay committed:
Fix url lookup test to use test container (#80284)

felixfontein committed:
Add support for plugin field in seealso. (#80212)

hswong3i committed:
ansible-galaxy - support `resolvelib >= 0.5.3, < 1.1.0` (#80196)

* ansible-galaxy - support `resolvelib >= 0.5.3, < 1.1.0`

<https://pypi.org/project/resolvelib/1.0.1> released on 2023-03-09:

-   <https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst#101-2023-03-09>
-   <https://github.com/sarugaku/resolvelib/releases/tag/1.0.1>

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

* Trigger CI by pinning resolvelib with latest version

Since resolvelib is pre-installed in our test containers, we should temporarily pin the latest version allowed to force the tests to run with that version. Once the tests have passed that commit can be reverted.

Please make those changes without force pushing, so that we keep the reference to the passing CI run. We can squash the commits when merging the PR so the temporary commits won't be in the final commit merged to the devel branch.

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

* https://github.com/ansible/ansible/pull/80196#discussion_r1136003637

Also test resolvelib with multiple supported versions.

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

* Revert "Trigger CI by pinning resolvelib with latest version"

This reverts commit 5518e5dbca01b87d13850cee9ccdbdc1ee49188f.

---------

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

bcoca committed:
password lookup, handle ident properly when saved (#80251)

* password lookup, handle ident properly when saved

  Currently we format and save ident when present but we didn't account for this when reading the saved file
  Also added some more robust error handling.

mattclay committed:
Use variable instead of container resource in AZP (#80299)

ziegenberg committed:
Improve dirname and basename filter doc (#80054)

mattclay committed:
New upstream release tool (#80179)

mattclay committed:
ansible-test - Remove macos/12.0 remote (#80319)

jborean93 committed:
Ansible.Basic - Improve temporary file cleanup process (#80293)

* Ansible.Basic - Improve temporary file cleanup process

* Add comment on struct value used

felixfontein committed:
Fix URL processing. (#80295)

felixfontein committed:
Bump antsibull-docs version in docs-build sanity test (#80326)

* Bump antsibull-docs to 1.11.0.

* Update all requirements for the docs-build sanity test.

* Run again with Python 3.9.

jborean93 committed:
Improve Ansible.Basic.cs tempdir uniqueness (#80328)

* Improve Ansible.Basic.cs tempdir uniqueness

The current tempdir naming scheme can result in the same name if the
remote worker starts at the same time as another. By using the process
id it should add enough uniqueness to avoid this situation.

* Fix sanity issues

* Fix up compile issue on older hosts

4ch1m committed:
"Ansible Lint" IntelliJ plugin added (#80263)

bcoca committed:
clog (#80336)

(cherry picked from commit 97c8da77838e93b416c5e700abafa89a8383e950)

sivel committed:
Enforce that `Display.display` requires a `str` (#80327)

* Enforce that Display.display requires a str

* clog frag

* ci_complete

mattclay committed:
Add work-around for old AZP jobs (#80339)

anweshadas committed:
Add Ansible community 7.4.0 porting guide (#80338)

sivel committed:
New deb822_repository module (#80018)

Fixes #77073

mattclay committed:
Remove obsolete integration test aliases (#80342)

* Remove obsolete integration test aliases

* Remove obsolete `skip/osx` aliases

* Use `skip/macos` instead of `skip/osx` in docs

mattclay committed:
Use `skip/macos` instead of `skip/macos/13.2` (#80343)

The reason for the skip won't be going away with future versions of macOS.

mattclay committed:
Update package-data sanity test (#80344)

The test no longer relies on the Makefile.

mattclay committed:
Use a dedicated version helper for the docs build (#80345)

This removes the dependency on the release version helper in the docs-build sanity test.

mattclay committed:
Remove obsolete MANIFEST.in entries (#80346)

The referenced file and directory no longer exist.

felixfontein committed:
Add antsibull-changelog and antsibull-docs to other tools and programs page. (#80340)

mattclay committed:
Remove obsolete release bits (#80347)

Releases are now built using the `packaging/release.py` tool.
This makes the `Makefile` and associated files in `packaging/release/` and `packaging/sdist/` obsolete.

mattclay committed:
ansible-test - Update sanity test requirements (#80354)

mattclay committed:
ansible-test - Update base and default container (#80355)

mattclay committed:
ansible-test - Fix packaging change classification (#80360)

webknjaz committed:
🐛📝 Undouble trailing colon @ `faq.rst` (#80359)

It is causing parsing/representation problems of the following
explicit RST code block.

s-hertel committed:
fix using templated values for include/import role FA (#80320)

* fix using templated values for include/import role options 'public', 'allow_duplicates', and 'rolespec_validate'

* pass templated values without changing the instance

* Fix templating by setting always_post_validate to True and calling IncludeRole.post_validate() instead

ci_complete

* add changelog

mkrizek committed:
Add new dnf5 module (#80272)

felixfontein committed:
Support role extension for semantic markup. (#80305)

0ekk committed:
Fix run_once by instantly tiny post_validate (#78492) (#80051)

Signed-off-by: tu1h <lihai.tu@daocloud.io>

bcoca committed:
list all tags (including never) when listing tags (#80309)

* Ensure we default to show all tags when listing tags

 'never' was being excluded by default,as it is not part of the 'run tags'
 runtime default ('all'). For listing we now add it to the default 'run tags'.

mattclay committed:
Use --no-isolation for package-data sanity test (#80377)

The dependencies are already in the sanity test venv. This avoids use of unpinned dependencies and a dependency on a network connection.

felixfontein committed:
Implement semantic markup support for Ansible documentation in validate-modules. (#80243)

nitzmahone committed:
bump devel to 2.16.0.dev0 (#80390)
